stetho/src/main/java/com/facebook/stetho/inspector/elements/UIDOMProvider.java

```diff
@@ -0,0 +1,101 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+package com.facebook.stetho.inspector.elements;
+
+import android.app.Activity;
+import android.app.Application;
+import android.content.Context;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.Window;
+import android.widget.TextView;
+
+import com.facebook.stetho.inspector.protocol.module.DOM;
+
+public final class UIDOMProvider implements DOM.Provider {
+  private final NodeDescriptorMap mNodeDescriptorMap;
+  private final Application mApplication;
+  private Listener mListener;
+
+  public UIDOMProvider(Context context) {
+    mApplication = (Application)context.getApplicationContext();
```

Comment:
This is a pretty dangerous assumption to make at such a low level.  Strongly prefer requiring the caller pass in `Application` (which moves the assumption higher up the stack).  This will make it easier for us to correct it by actually requiring the `Application` object at some point.